install/kubernetes/cilium/templates/cilium-configmap.yaml

```diff
@@ -84,6 +85,14 @@
   {{- $defaultK8sClientQPS = 10 -}}
   {{- $defaultK8sClientBurst = 20 -}}
 {{- end -}}
+
+{{- if or (.Values.gke.enabled) (eq .Values.routingMode "native") -}}
+  {{- $tunnelProto = "" -}}
+{{- else if ne (.Values.tunnelProtocol) "" -}}
+  {{- $tunnelProto = .Values.tunnelProtocol -}}
+{{- else -}}
+  {{- $tunnelProto = "vxlan" -}}
+{{- end -}}
```

Comment:
A couple of drive-by comments:
* the tunnel protocol shall be configurable also in native routing mode. Indeed, there are certain features (e.g., Egress Gateway, DSR) which still create a tunnel for certain traffic categories in that case.
* the `tunnelProto` variable is currently never used, hence all this logic is not taking effect.